include/flags.h

```diff
@@ -0,0 +1,29 @@
+#ifndef FLAGS_H
+#define FLAGS_H
+
+#include <iostream>
+#include <map>
+#include <climits>
+
+class Flags {
+private:
+  int m_argc;
+  char** m_argv;
+
+  struct data {
+    unsigned char type; // 1: string, 2: int, 3: bool
+    std::string content;
+  };
+  std::map<std::string,data> parsed;
+
+  Flags(){};
```

Comment:
Why are you overriding the default constructor? You've provided a custom constructor below, so this is already overridden.